apps/emqx_bridge_mqtt/src/emqx_bridge_mqtt_connector.erl

```diff
@@ -481,8 +482,17 @@ mk_client_opts(
         msg_handler => mk_client_event_handler(Name, TopicToHandlerIndex)
     }.
 
-mk_clientid(WorkerId, ClientId) ->
-    emqx_bridge_mqtt_lib:bytes23([ClientId], WorkerId).
+mk_clientid(WorkerId, {Prefix, ClientId}) when ?IS_NO_PREFIX(Prefix) ->
+    %% When there is no prefix, try to keep the client ID length within 23 bytes
+    emqx_bridge_mqtt_lib:bytes23(ClientId, WorkerId);
+mk_clientid(WorkerId, {Prefix, ClientId}) when size(Prefix) < 20 ->
+    %% Try to respect client ID prefix when it's less than 20 bytes
+    %% meaning there is at least 3 bytes to randomize
+    %% Must add $: for backward compatibility
+    emqx_bridge_mqtt_lib:bytes23_with_prefix(Prefix, ClientId, WorkerId);
```

Comment:
Q: Can we also involve `proto_ver` as configured by the user in the decision making? There probably should be no such weird limitation if it's `v5`?

Reply:
Frim v5 spec:
> The Server MAY allow ClientID’s that contain more than 23 encoded bytes. The Server MAY allow ClientID’s that contain characters not included in the list given above.